### ISSUE TITLE
[Bug] TUN device does not accept IPv6 traffic, even if the routes are added manually. Willing to help.

### ISSUE THREAD

**Issue by imkiva** [Dreamacro/clash#2932]
### Prerequisites

- [X] I understand that this is the official open-source version of the Clash core, **only providing support for the open-source version or Premium version**
- [X] I am submitting an issue with the Clash core, not Clash.Meta / OpenClash / ClashX / Clash For Windows or any other derivative version
- [X] I am using the latest version of the Clash or Clash Premium core **in this repository**
- [X] I have searched at the [Issue Tracker](……/) **and have not found any related issues**
- [X] I have read the [official Wiki](https://dreamacro.github.io/clash/) **and was unable to solve the issue**
- [X] (required for Premium core) I've tried the `dev` branch and the issue still exists

### Version

2023.08.17-13-gdcc8d87 (the 5/9/2023 build from S3 storage)

### Operating System

Linux

### Architecture

arm64

### Configuration File

```YAML
socks-port: 7891
allow-lan: true
mode: Rule
ipv6: true
tun:
  enable: true
  stack: system
  auto-route: false
dns:
  listen: :10053
  enable: true
  enhanced-mode: fake-ip
  ipv6: true
  nameserver:
    - 119.29.29.29
    - 223.5.5.5
  fake-ip-range: 198.18.0.1/16
```

## TL;DR

After some hours of searching the web, I realized that the tun device created by clash [only accepts IPv4 traffics](https://github.com/Dreamacro/clash/blob/d034a408be42815e98f3aea80be24949946aea83/docs/premium/tun-device.md?plain=1#L13).

**I would like to help with this if maintainers have something more important to schedule.** In that case, I can get rid of one extra tool that bridges between clash and IPv6 traffic. (Of couse I can use tproxy, but it does not elegent if it is used with OSPF). 
But unfortunately I don't have access to the source code related to tun mode. So you may treat this issue as a permission request more than a bug report.

FWIW, I know when Fake IP is enabled, whether the proxy-ed connection will access target website though IPv6 **has nothing to do with** if TUN accepts IPv6 or not. I know it is only up to whether the proxy server has an IPv6 connection. What I want to say is, I want to access some service though proxy, but the service only offers IPv6 connection and no domain name was resolved to that address, in which case, the tun device should accept IPv6 and send it to the proxy server.

## Repro

My setup:
- Clash: DNS, Fake IP, and tun mode are enabled.
- `tun.auto-route` is set to `false` on purpose (described later).
- A private IPv6 address is manually assigned to clash tun device `utun`, by:

```shell
# ip -6 addr add fdfe:dcba:9876::2/126 dev utun
``` 

The routing table is managed and then advertised by OSPF protocol, like:

```shell
# ip route | head -n 5
default via 192.168.255.1 dev eth0 src 192.168.255.254 metric 100
198.18.0.0/16 dev utun proto kernel scope link src 198.18.0.1

# ip -6 route | head -n 5
::1 dev lo proto kernel metric 256 pref medium
fe80::/64 dev utun proto kernel metric 256 pref medium
```

The tun device works great for almost all cases. Except when a new IPv6 route is advertised and I want to connect to that address directly (no name resolution), say, for example:
```shell
# ip -6 route
<... omitted ...>
2408:4001:f00/48 dev utun proto bird metric 1024 pref medium
```

I am unable to connect to the target address by:
```shell
# curl -k --interface utun https://[2408:4001:f00::27c]
<... curl stucks here, no response ...>
```

And there's no related log shown in clash dashboard.

## Workaround
If I create a new tun device named `utun6` with `tun2socks`, which connects to the clash socks proxy:
```shell
# ip tuntap add dev utun6 mod tun
# ip -6 addr add fdfe:dcba:9876::1/126 dev utun6
# tun2socks -mtu 9000 -device utun6 -proxy socks5://[::1]:7891
```

and update the IPv6 routing table, by replacing every "utun" with "utun6". The result should be like:

```
# ip -6 route
2408:4001:f00/48 dev utun6 proto bird metric 1024 pref medium
```

Finally, `curl` can successfully fetch the content on that address:

```
# curl -k --interface utun6 https://[2408:4001:f00::27c]
<...  HTML contents ...>
```

And in clash dashboard, the log page shows lines like:

```
[TCP] connected lAddr=[::1]:55492 rAddr=[2408:4001:f00::27c]:443 mode=rule <omitted>
```

### Reproduction Steps

As described above :)

**Comment by Dreamacro** [comment]
AFAIN, @nekohasekai working on it

**Comment by imkiva** [comment]
Thanks. Care to share some progress? Or anything that I can help with? --- I just can't wait for it.
